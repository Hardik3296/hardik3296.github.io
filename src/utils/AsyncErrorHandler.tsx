import React from "react";

const useAsyncError = () => {
   const [_error, setError] = React.useState();
   return React.useCallback(
      e => {
         setError(() => {
            throw e;
         });
      },
      [setError],
   );
};

export default useAsyncError;