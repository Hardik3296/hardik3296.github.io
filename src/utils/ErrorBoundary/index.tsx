import { Component, ReactNode } from "react";
import styles from "./styles.module.scss";

interface MyProps {
   textcolor: string,
   children: ReactNode,
}

class ErrorBoundary extends Component<MyProps, {}> {

   state = {
      hasError: false
   }

   static getDerivedStateFromError() {
      return { hasError: true }
   }

   componentDidCatch(error: any, errorInfo: any) {
   }
   render() {
      if (this.state.hasError)
         return <div className={styles.errorDiv}>
            <p style={{ color: this.props.textcolor }}>Something went wrong while fetching the repositories!!</p>
         </div>
      else
         return this.props.children;
   }
}

export default ErrorBoundary;