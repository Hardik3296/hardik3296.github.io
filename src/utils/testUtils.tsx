import { ShallowWrapper } from "enzyme"

export const findAttributeByTestId = (component:ShallowWrapper, attr:string):ShallowWrapper=>{
   return component.find(`[test-id='${attr}']`);
}