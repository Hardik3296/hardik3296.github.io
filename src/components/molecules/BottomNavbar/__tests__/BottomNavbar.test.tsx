import BottomNavBar from "..";
import {shallow, ShallowWrapper} from "enzyme";
import renderer from "react-test-renderer"
import {findAttributeByTestId} from "../../../../utils/testUtils";

const renderComponent=():ShallowWrapper=>{
   return shallow(<BottomNavBar/>)
}

describe("Bottom Navbar",()=>{
   let wrapper:ShallowWrapper;

   beforeEach(()=>{
      wrapper=renderComponent();
   });

   it("Should have one profile anchor",()=>{
      const component = findAttributeByTestId(wrapper,"me");
      expect(component.length).toBe(1);
   });

   it("Should have one Education anchor",()=>{
      const component = findAttributeByTestId(wrapper,"Education");
      expect(component.length).toBe(1);
   });

   it("Should have one WorkEx anchor",()=>{
      const component = findAttributeByTestId(wrapper,"WorkEx");
      expect(component.length).toBe(1);
   });

   it("Should have one GithubRepos anchor",()=>{
      const component = findAttributeByTestId(wrapper,"GithubRepos");
      expect(component.length).toBe(1);
   });

   it("Should have one theme icon",()=>{
      const component = findAttributeByTestId(wrapper,"ThemeIcon");
      expect(component.length).toBe(1);
   });

   it("renders correctly",()=>{
      const snapshot = renderer.create(<BottomNavBar/>).toJSON();
      expect(snapshot).toMatchSnapshot();
   });

});