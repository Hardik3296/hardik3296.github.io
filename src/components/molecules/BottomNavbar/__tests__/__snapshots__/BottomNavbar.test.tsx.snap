// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bottom Navbar renders correctly 1`] = `
<div
  className="bottomNavigation"
  style={
    Object {
      "backgroundColor": "#f2f2f2",
    }
  }
>
  <div
    onClick={[Function]}
    test-id="me"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root person"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
  <div
    onClick={[Function]}
    test-id="Education"
  >
    <img
      alt=""
      className="imageIcons"
      src="study.png"
    />
  </div>
  <div
    onClick={[Function]}
    test-id="WorkEx"
  >
    <img
      alt=""
      className="imageIcons"
      src="new-job.png"
    />
  </div>
  <div
    onClick={[Function]}
    test-id="GithubRepos"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root github"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
      />
    </svg>
  </div>
  <div
    className="bottomIconDiv"
    onClick={[Function]}
    test-id="ThemeIcon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root sunny"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M6.76 4.84l-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91l-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7l1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91l1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
      />
    </svg>
  </div>
</div>
`;
